docs/remote/containers.md

```diff
@@ -869,18 +883,24 @@ You can build images and deploy containers by forwarding the Docker socket and i
 
 ### What are the connectivity requirements for the VS Code Server when it is running in a container?
 
-The VS Code Server requires outbound HTTPS (port 443) connectivity to:
+Installation of VS Code Server requires that your local machine have outbound HTTPS (port 443) connectivity to:
 
 * `update.code.visualstudio.com`
-* `marketplace.visualstudio.com`
 * `vscode.blob.core.windows.net`
 * `*.vo.msecnd.net` (Azure CDN)
-* `*.gallerycdn.vsassets.io` (Azure CDN)
 
-All other communication between the server and the VS Code client is accomplished through an authenticated, random, TCP port automatically exposed via the Docker CLI. You can find a list of locations VS Code itself needs access to in the [network connections article](/docs/setup/network.md#common-hostnames).
+The Remote - Containers extensions will download VS Code Server locally and copy it to the container once connected.
+
+You can install extensions manually without an internet connection using the **Extensions: Install from VSIX...** command, but if you use the extension panel or `devcontainer.json` to install extensions, your local machine and VS Code Server will need outbound HTTPS (port 443) access  to:
+
+* `marketplace.visualstudio.com`
+* `*.vo.msecnd.net` (Azure CDN)
+* `*.gallerycdn.vsassets.io` (Azure CDN)
 
 Finally, some extensions (like C#) download secondary dependencies from `download.microsoft.com` or `download.visualstudio.microsoft.com`. Others (like [Visual Studio Live Share](https://docs.microsoft.com/visualstudio/liveshare/reference/connectivity#requirements-for-connection-modes)) may have additional connectivity requirements. Consult the extension's documentation for details if you run into trouble.
 
+All other communication between the server and the VS Code client is accomplished through an authenticated, random port automatically exposed via the Docker CLI.
```

Comment:
The default is a socket path. There is a default port (not a random one) when you enable listening on a port for the daemon, but by default there is no port open and the port is not secured by default.

Reply:
Docker uses a socket path by default locally. The VS Code Server always uses a port inside the container on the container's local network interface that is not available on the outside (unless the container shares the network with the host, I think). We are using `docker exec` to go through the socket path and connect to the port inside the container. The port inside the container is a random one, that's correct. Maybe this is more information than the user needs though.